### ISSUE TITLE
wandb metric argument is weird

### ISSUE THREAD

**Issue by edmcman** [huggingface/transformers#24409]
### System Info

- `transformers` version: 4.30.2
- Platform: Linux-5.7.19-050719-generic-x86_64-with-glibc2.29
- Python version: 3.8.10
- Huggingface_hub version: 0.15.1
- Safetensors version: 0.3.1
- PyTorch version (GPU?): 2.0.1+cu117 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <yes>
- Using distributed or parallel set-up in script?: <no>

### Who can help?

@AyushExel

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

There are a few bugs/weirdness when specifying the `metric` in `wandb` hyperparameter search.

1. If a `metric` is specified via the `hp_space` argument but there is not a `metric` argument to `hyperparameter_search`, the metric in the `hp_space` is ignored and changed to `eval/loss`.  See: https://github.com/huggingface/transformers/blame/6ce6d62b6f20040129ec9831e7c4f6576402ea42/src/transformers/integrations.py#L497
2. If a custom `hp_space` is provided that does not define `metric` at all, and the `metric` argument is specified to `hyperparameter_search`, this code throws an exception: https://github.com/huggingface/transformers/blame/6ce6d62b6f20040129ec9831e7c4f6576402ea42/src/transformers/integrations.py#L500 

### Expected behavior

1. If a `hp_space` defines a metric, use that instead of overwriting it.
2. Don't throw an exception is the `hp_space` lacks a `metric` key.

**Comment by AyushExel** [comment]
@ayulockin @morganmcg1 in case this is relevant to the integrations team. tbh, I don't remember much about this integration but I implemented it so happy to help in case there's a blocker.

**Comment by ayulockin** [comment]
Thanks for the detailed issue @edmcman; we are taking a look at this. Will connect with you @AyushExel in case of blocker.

**Comment by edmcman** [comment]
Ping

On Mon, Jul 31, 2023, 11:03 AM github-actions[bot] -
***@***.*** <github.edmcman.99c9f1b9d0.notifications#

**Comment by edmcman** [comment]
Ping

**Comment by edmcman** [comment]
Ping

**Comment by edmcman** [comment]
No worries, I was just keeping the issue open :)

**Comment by edmcman** [comment]
Boop

**Comment by edmcman** [comment]
Boop

**Comment by edmcman** [comment]
Boop

**Comment by edmcman** [comment]
Bump

**Comment by edmcman** [comment]
Boop

**Comment by amyeroberts** [comment]
Hi @AyushExel - any update on this?

**Comment by edmcman** [comment]
@ayulockin can you help?

**Comment by edmcman** [comment]
.

**Comment by amyeroberts** [comment]
cc @parambharat as they've been doing loads of great integration work between HF & W&B recently
